examples/hsgp.py

```diff
@@ -0,0 +1,711 @@
+        day_effect = 100 * day_effect.mean(axis=0)
+
+        fig = plt.figure(figsize=(12, 5))
+        plt.plot(np.arange(1, 367), day_effect)
+        for name, day in special_days.items():
+            xs = day.day_of_year
+            ys = day_effect[day.day_of_year - 1]
+            plt.plot(xs, ys, marker="o", mec="k", c="none", ms=10)
+            plt.text(xs - 3, ys, name, horizontalalignment="right")
+        plt.title("Special day effect")
+        plt.ylabel("Relative number of births")
+        plt.xlabel("Day of year")
+        plt.xlim([-40, None])
+        return fig
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Hilbert space approx for GPs")
+    parser.add_argument("--num-samples", nargs="?", default=1000, type=int)
+    parser.add_argument("--num-warmup", nargs="?", default=1000, type=int)
+    parser.add_argument("--num-chains", nargs="?", default=1, type=int)
+    parser.add_argument(
+        "--model",
+        nargs="?",
+        default="tywd",
+        help="one of"
+        '"t" (Long term trend),'
+        '"ty" (t + year seasonality),'
+        '"tyw" (t + y + slowly varying weekday effect),'
+        '"tywd" (t + y + w + special days effect)',
+    )
+    parser.add_argument("--device", default="cpu", type=str, help='use "cpu" or "gpu".')
+    parser.add_argument("--x64", action="store_true", help="Enable float64 precision")
+    parser.add_argument(
+        "--save-samples",
+        default="",
+        type=str,
+        help="Path where to store the samples. Must be '.npz' file.",
+    )
+    parser.add_argument(
+        "--save-figure",
+        default="",
+        type=str,
+        help="Path where to save the plot with matplotlib.",
+    )
+    args = parser.parse_args()
+    return args
+
+
+NAME_TO_MODEL = {
+    "t": GP1,
+    "ty": GP2,
+    "tyw": GP3,
+    "tywd": GP4,
+}
+
+
+def main(args):
+    model = NAME_TO_MODEL[args.model]()
+    data = model.get_data()
+    print(f"x: mean={model.x_scaler._mean}, sd={model.x_scaler._std}")
+    print(f"y: mean={model.y_scaler._mean}, sd={model.y_scaler._std}")
+    mcmc = MCMC(
+        NUTS(model.model),
+        num_warmup=args.num_warmup,
+        num_samples=args.num_samples,
+        num_chains=args.num_chains,
+        progress_bar=False if "NUMPYRO_SPHINXBUILD" in os.environ else True,
+    )
+    mcmc.run(jax.random.PRNGKey(0), **data)
+    mcmc.print_summary()
+    posterior_samples = mcmc.get_samples()
+    if args.save_samples:
+        print(f"Saving samples at {args.save_samples}")
+        save_samples(args.save_samples, posterior_samples)
+    if args.save_figure:
+        print(f"Saving figure at {args.save_figure}")
+        fig = model.make_figure(posterior_samples)
+        fig.savefig(args.save_figure)
+        plt.close()
+
+    return model, data, mcmc, posterior_samples
+
+
+if __name__ == "__main__":
+    args = parse_arguments()
+    jax.config.update("jax_enable_x64", args.x64)
```

Comment:
you can also do `jax_enable_x64`

Reply:
Nice, thanks for this tip. I've updated the PR.